### PR TITLE
JAMES-3078 ADR for Reactor-netty adoption for JMAP server implementation

### PR DESCRIPTION
**Pull request by Arsnael**

### CODE REVIEW

File: src/adr/0019-reactor-netty-adoption.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+After we decided to adopt and implement the last specifications of JMAP (see 
+[new JMAP specifications adoption ADR](https://github.com/apache/james-project/blob/master/src/adr/0018-jmap-new-specs.md)), 
+it was agreed that we would need to be able to serve both `jmap-draft` and the new `jmap` with a reactive server. 
+
+## Decision
+
+We decided to use [reactor-netty](https://github.com/reactor/reactor-netty) for the following reasons:
+
+* it's a reactive server
+* it's using [Reactor](https://projectreactor.io/), which is the same technology that we use in the rest of our codebase
```
Comment (chibenwa) [inline src/adr/0019-reactor-netty-adoption.md]:
```
 * Implementing JMAP do not require high level HTTP server features
```

File: src/adr/0019-reactor-netty-adoption.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+[new JMAP specifications adoption ADR](https://github.com/apache/james-project/blob/master/src/adr/0018-jmap-new-specs.md)), 
+it was agreed that we would need to be able to serve both `jmap-draft` and the new `jmap` with a reactive server. 
+
+## Decision
+
+We decided to use [reactor-netty](https://github.com/reactor/reactor-netty) for the following reasons:
+
+* It's a reactive server
+* It's using [Reactor](https://projectreactor.io/), which is the same technology that we use in the rest of our codebase
+* Implementing JMAP do not require high level HTTP server features
```
Comment (rouazana) [inline src/adr/0019-reactor-netty-adoption.md]: does

File: src/adr/0019-reactor-netty-adoption.md [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+## Status
+
+Accepted (lazy consensus)
+
+## Context
+
+After we decided to adopt and implement the last specifications of JMAP (see 
+[new JMAP specifications adoption ADR](https://github.com/apache/james-project/blob/master/src/adr/0018-jmap-new-specs.md)), 
+it was agreed that we would need to be able to serve both `jmap-draft` and the new `jmap` with a reactive server.
```
Comment (mbaechler) [inline src/adr/0019-reactor-netty-adoption.md]: We should write ADR at the present tense. Here "We decide"

File: src/adr/0019-reactor-netty-adoption.md [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+Accepted (lazy consensus)
+
+## Context
+
+After we decided to adopt and implement the last specifications of JMAP (see 
+[new JMAP specifications adoption ADR](https://github.com/apache/james-project/blob/master/src/adr/0018-jmap-new-specs.md)), 
+it was agreed that we would need to be able to serve both `jmap-draft` and the new `jmap` with a reactive server. 
+
+## Decision
```
Comment (mbaechler) [inline src/adr/0019-reactor-netty-adoption.md]: What are alternatives? why do we want to ditch jetty?

File: src/adr/0019-reactor-netty-adoption.md [hunk 5]
```diff
@@ -8,21 +8,30 @@ Accepted (lazy consensus)
...
 * It's a reactive server
 * It's using [Reactor](https://projectreactor.io/), which is the same technology that we use in the rest of our codebase
 * Implementing JMAP do not require high level HTTP server features
 
 ## Consequences
 
-* Porting current `jmap-draft` to use a `reactor-netty` server instead of a [Jetty server](https://www.eclipse.org/jetty/)
+* Porting current `jmap-draft` to use a `reactor-netty` server instead of a Jetty server
 * The `reactor-netty` server should serve as well the new `jmap` implementation
 * Should be possible to refactor and get end-to-end reactive operations with it
```
Comment (chibenwa) [inline src/adr/0019-reactor-netty-adoption.md]: * We will be able to refactor and get end-to-end reactive operations for JMAP, unlocking performance gains

**Comment by Arsnael** [comment]
Merged

**Pull request closed without merging**